src/hotspot/share/opto/loopopts.cpp

```diff
@@ -352,13 +352,14 @@ void PhaseIdealLoop::dominated_by(IfProjNode* prevdom, IfNode* iff, bool flip, b
     if (cd->depends_only_on_test() && _igvn.no_dependent_zero_check(cd)) {
       assert(cd->in(0) == dp, "");
       _igvn.replace_input_of(cd, 0, prevdom);
-      if (range_check_predicate) {
-        // Loads and range check Cast nodes that are control dependent on this range check (that is about to be removed)
-        // now depend on multiple dominating range checks. After the removal of this range check, these control
-        // dependent nodes end up at the lowest/nearest dominating check in the graph. To ensure that these Loads/Casts
-        // do not float above any of the dominating checks (even when the lowest dominating check is later replaced by
-        // yet another dominating check), we need to pin them at the lowest dominating check.
-        Node* clone = cd->pin_for_array_access();
+      if (pin_array_nodes) {
+        // Because of range check predication, Loads and range check Cast nodes that are control dependent on this range
```

Comment:
Loop Predication?
```suggestion
        // Because of Loop Predication, Loads and range check Cast nodes that are control dependent on this range
```